### PR TITLE
fix: support whitespace and signed numbers in parser

### PR DESCRIPTION
**Pull request by blackboxprogramming**
## Summary
- broaden `parse_numeric_prefix` to accept whitespace and +/- signs using regex
- add tests for spaced and negative prefixes

## Testing
- `pre-commit run --files prism_utils.py tests/test_prism_utils.py main.py`
- `pytest tests/test_prism_utils.py`

------
https://chatgpt.com/codex/tasks/task_e_68c32fe0713883298b049163f0883020

### CODE REVIEW

File: prism_utils.py [hunk 1]
```diff
@@ -3,19 +3,24 @@
 from __future__ import annotations
 
 import ast
+import re
+
+_NUMERIC_PREFIX_RE = re.compile(r"^\s*([+-]?(?:\d+(?:\.\d*)?|\.\d+))")
```
[the remark this answers is not in the document]
Comment (blackboxprogramming) [reply prism_utils.py]: @codex fix comments please
